src/histogram.c

```diff
@@ -135,65 +295,86 @@ hist_combinefunc(PG_FUNCTION_ARGS)
 Datum
 hist_serializefunc(PG_FUNCTION_ARGS)
 {
-	bytea	   *state;
-	Datum	   *hist;
-	Size		i;
+	HistogramState	   *state;
+	StringInfoData		buf;
 
 	Assert(!PG_ARGISNULL(0));
-	state = PG_GETARG_BYTEA_P(0);
-	hist = (Datum *) VARDATA(state);
+	state = (HistogramState *) PG_GETARG_POINTER(0);
+	pq_begintypsend(&buf);
+	pq_sendint(&buf, VARSIZE(state->values) - VARHDRSZ, 4);
+	pq_sendbytes(&buf, VARDATA(state->values), VARSIZE(state->values) - VARHDRSZ);
 
-	for (i = 0; i < HIST_LEN(state); i++)
-	{
-		hist[i] = Int32GetDatum(htonl(DatumGetInt32(hist[i])));
-	}
+	pq_sendint(&buf, VARSIZE(state->opt_values) - VARHDRSZ, 4);
+	pq_sendbytes(&buf, VARDATA(state->opt_values), VARSIZE(state->opt_values) - VARHDRSZ);
```

Comment:
When using these serialization functions, I think we can get rid of the winsock lib dependency for the windows build that was needed due to `htonl`.